framework/validators/CEmailValidator.php

```diff
@@ -156,15 +162,24 @@ protected function checkMxPorts($domain)
 		$records=dns_get_record($domain, DNS_MX);
 		if($records===false || empty($records))
 			return false;
+		if (is_int($this->timeout)) {
+			$timeout=$this->timeout;
+		}else {
+			$timeout=((int)ini_get("default_socket_timeout"))*1000;
+		}
 		usort($records,array($this,'mxSort'));
 		foreach($records as $record)
 		{
-			$handle=@fsockopen($record['target'],25);
-			if($handle!==false)
+			$url='smtp://'.$record['target'].':25/';
+			$ch=curl_init($url);
```

Comment:
I do not see the benefit of configuring timeout in milliseconds here? Configuring a timeout under 1 second can cause problems with slowly responding mailservers so you would reject email from people which have slow servers or which have high latency connections. I prefer to use timeout parameter of fsockopen instead of introducing the dependency on curl here.